transport.go

```diff
@@ -20,9 +20,14 @@ import (
 
 const RaftProtocol protocol.ID = "/raft/1.0.0/rpc"
 
+// Ensure that streamLayer implements raft.StreamLayer.
+var _ raft.StreamLayer = (*streamLayer)(nil)
+
+// Ensure that HCLogToLogger implents hclog.Logger.
+var _ hclog.Logger = (*HcLogToLogger)(nil)
```

Comment:
This is technically not necessary but thought it may be a good idea to force interface compatibility at compile time.